server/src/main/java/org/elasticsearch/common/settings/SettingsModule.java

```diff
@@ -210,6 +210,13 @@ private void registerSetting(Setting<?> setting) {
         }
     }
 
+    // TODO: remove this hack once we remove the deprecated ability to use repository settings in the cluster state in the S3 snapshot
+    // module
+    private boolean isS3InsecureCredentials(Setting<?> setting) {
```

Comment:
This is a weird hack admittedly, but I needed this to get the ability to register the settings for filtering again. We don't have the requirement about the name containing a dot in 7.x and since this is just temporary I figured it was good enough.